web/secondary_toolbar.js

```diff
@@ -29,173 +29,232 @@
-    this.eventBus = eventBus;
+/**
+ * @class
+ */
+var SecondaryToolbar = (function SecondaryToolbarClosure() {
+  /**
+   * @constructs SecondaryToolbar
+   * @param {SecondaryToolbarOptions} options
+   * @param {EventBus} eventBus
+   */
+  function SecondaryToolbar(options, eventBus) {
     this.toolbar = options.toolbar;
-    this.buttonContainer = this.toolbar.firstElementChild;
-
-    // Define the toolbar buttons.
     this.toggleButton = options.toggleButton;
     this.presentationModeButton = options.presentationModeButton;
-    this.openFile = options.openFile;
-    this.print = options.print;
-    this.download = options.download;
-    this.viewBookmark = options.viewBookmark;
-    this.firstPage = options.firstPage;
-    this.lastPage = options.lastPage;
-    this.pageRotateCw = options.pageRotateCw;
-    this.pageRotateCcw = options.pageRotateCcw;
-    this.toggleHandTool = options.toggleHandTool;
+    this.openFileButton = options.openFile;
+    this.printButton = options.print;
+    this.downloadButton = options.download;
+    this.viewBookmarkButton = options.viewBookmark;
+    this.firstPageButton = options.firstPage;
+    this.lastPageButton = options.lastPage;
+    this.pageRotateCwButton = options.pageRotateCw;
+    this.pageRotateCcwButton = options.pageRotateCcw;
+    this.toggleHandToolButton = options.toggleHandTool;
     this.documentPropertiesButton = options.documentPropertiesButton;
 
-    // Attach the event listeners.
-    var elements = [
-      // Button to toggle the visibility of the secondary toolbar:
-      { element: this.toggleButton, handler: this.toggle },
-      // All items within the secondary toolbar
-      { element: this.presentationModeButton,
-        handler: this.presentationModeClick },
-      { element: this.openFile, handler: this.openFileClick },
-      { element: this.print, handler: this.printClick },
-      { element: this.download, handler: this.downloadClick },
-      { element: this.viewBookmark, handler: this.viewBookmarkClick },
-      { element: this.firstPage, handler: this.firstPageClick },
-      { element: this.lastPage, handler: this.lastPageClick },
-      { element: this.pageRotateCw, handler: this.pageRotateCwClick },
-      { element: this.pageRotateCcw, handler: this.pageRotateCcwClick },
-      { element: this.toggleHandTool, handler: this.toggleHandToolClick },
-      { element: this.documentPropertiesButton,
-        handler: this.documentPropertiesClick }
-    ];
-
-    for (var item in elements) {
-      var element = elements[item].element;
-      if (element) {
-        element.addEventListener('click', elements[item].handler.bind(this));
+    this.eventBus = eventBus;
+
+    this.opened = false;
+    this.previousContainerHeight = null;
+    this.newContainerHeight = null;
+    this.buttonContainer = this.toolbar.firstElementChild;
+
+    // Bind the event listeners for click and hand tool actions.
+    this.bindClickListeners();
+    this.bindHandToolListener();
```

Comment:
I'll try to do a proper review later on, but just a drive-by comment so far: Let's make these methods "private", i.e. `this._bind...` (with the associate JSDoc comments).